src/bentoml/_internal/io_descriptors/text_stream.py

```diff
@@ -0,0 +1,163 @@
+
+    .. code-block:: bash
+
+       % bentoml serve ./service.py:svc --reload
+
+    Users can then send requests to the newly started services with any client:
+
+    .. tab-set::
+
+        .. tab-item:: Bash
+
+           .. code-block:: bash
+
+              % curl -X POST http://0.0.0.0:3000/count --d 'Hello World'
+
+    .. note::
+
+        :obj:`TextStream` is not designed to take any ``args`` or ``kwargs`` during initialization.
+
+    Returns:
+        :obj:`TextStream`: IO Descriptor that represents stream of string type.
+    """
+
+    _mime_type = MIME_TYPE
+
+    def __init__(self, *args: t.Any, **kwargs: t.Any):
+        if args or kwargs:
+            raise BentoMLException(
+                f"'{self.__class__.__name__}' is not designed to take any args or kwargs during initialization."
+            ) from None
+
+    def _from_sample(self, sample: str | bytes):
+        raise NotImplementedError("TextStream does not support input type for now")
+
+    def input_type(self) -> t.Type[str]:
+        raise NotImplementedError("TextStream does not support input type for now")
+
+    def to_spec(self) -> dict[str, t.Any]:
+        return {"id": self.descriptor_id}
+
+    @classmethod
+    def from_spec(cls, spec: dict[str, t.Any]) -> Self:
+        return cls()
+
+    def openapi_schema(self) -> Schema:
+        return Schema(type="string")
+
+    def openapi_components(self) -> dict[str, t.Any] | None:
+        pass
+
+    def openapi_example(self):
+        pass
+
+    def openapi_request_body(self) -> dict[str, t.Any]:
+        return {
+            "content": {
+                self._mime_type: MediaType(
+                    schema=self.openapi_schema(), example=self.openapi_example()
+                )
+            },
+            "required": True,
+            "x-bentoml-io-descriptor": self.to_spec(),
+        }
+
+    def openapi_responses(self) -> OpenAPIResponse:
+        return {
+            "description": SUCCESS_DESCRIPTION,
+            "content": {
+                self._mime_type: MediaType(
+                    schema=self.openapi_schema(), example=self.openapi_example()
+                )
+            },
+            "x-bentoml-io-descriptor": self.to_spec(),
+        }
+
+    async def from_http_request(self, request: Request) -> str:
+        raise NotImplementedError("TextStream does not support input type for now")
+
+    async def to_http_response(
+        self, obj: ContentStream, ctx: Context | None = None
+    ) -> StreamingResponse:
+        if ctx is not None:
+            res = StreamingResponse(
+                obj,
+                media_type=self._mime_type,
+                headers=ctx.response.metadata,  # type: ignore (bad starlette types)
+                status_code=ctx.response.status_code,
+            )
+            set_cookies(res, ctx.response.cookies)
+            return res
+        else:
+            return StreamingResponse(obj, media_type=self._mime_type)
```

Comment:
For SSE, we don't want a simple TextStream, but indeed an [EventStream](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#event_stream_format), it would be most appropriate to wrap it here. I've spotted this point on the meeting, but it seems we still need to align, cc @aarnphm @sauyon @ssheng

Reply:
I can't execute the request on the swagger page so I don't know it is implemented correctly or not. But theoretically, an valid event stream should be generated with mime type `text/event-stream`

Reply:
Just making sure that we are not speaking pass each other, you are suggesting that the returned object conform to the SSE data format, i.e. with `event` `data` `id` `retry` fields right? because the mime type is already set to `text/event-stream`

Reply:
Is there an issue with simply streaming text? I knowing we want to support "true" SSE we want to events, but for the simple text stream this should be fine, as long as we're not setting the mime incorrectly, right?

(If we are setting the mime to event-stream we should indeed be doing as frost says, though.)

Reply:
We are setting the mime to event-stream in this case? Even for text, I agree with frost here.

Reply:
I'm not really sure what you mean, I'm saying we can just change the mime type to text/plain and everything is fine as-is.

Reply:
> Just making sure that we are not speaking pass each other, you are suggesting that the returned text to conform to the SSE data format, i.e. with `event` `data` `id` `retry` fields right? because the mime type is already set to `text/event-stream`

Yes.

> I'm not really sure what you mean, I'm saying we can just change the mime type to text/plain and everything is fine as-is.

A text stream could be convenient only when user is interacting with the API. However, event stream is the standard protocol of SSE that is supported by main stream browsers, and event stream can carry more information in the event data. A possible solution is to add another type or an option to switch between plain text stream and event stream.

Reply:
One more thing to add, for building an event, it should have the flexibility to define the protocol. For example, Open AI API marks the end of stream by a `data: [DONE]` message, others could have different protocol.

Reply:
Would asking user to define the whole event be pushing too much complexity down to the user? would asking user to produce a generator of string and we wrap it with
`data: {string} \n\n` at the boundary works?

Reply:
More references,

[Claude's stream API](https://docs.anthropic.com/claude/reference/streaming#example) marks the end of stream with a non-empty `stop_reason` field
While [Open AI chat completion](finish_reason) does this with a nested `choices.finish_reason` field.

The protocols differ a lot. What I mean is to allow users to mark the end of stream with flexibility.

Reply:
> A text stream could be convenient only when user is interacting with the API. However, event stream is the standard protocol of SSE that is supported by main stream browsers, and event stream can carry more information in the event data. A possible solution is to add another type or an option to switch between plain text stream and event stream. After all the title "SSE" refers to an event stream, or it is called simply "streamed/chunked texts"

I think streamed responses predate SSE and support is widely available in JS for both; but yes, I agree that the title SSE is incorrect for this PR.